src/module.c

```diff
@@ -1023,11 +1023,19 @@ int RM_CreateSubcommand(RedisModuleCtx *ctx, const char *name, RedisModuleCmdFun
     return REDISMODULE_OK;
 }
 
+/* Returns 1 if `cmd` is a command of the module `modulename`. 0 otherwise. */
 int moduleIsModuleCommand(struct redisCommand *cmd, char *modulename) {
+    /* We do some very smple aching to save repeated calls to dictFetchValue
+     * (in case this function is called in a loop, with same `modulename` each time). */
+    static char *cached_modulename = NULL;
+    static struct RedisModule *cached_module = NULL;
     if (cmd->proc != RedisModuleCommandDispatcher)
         return 0;
-    // TODO:GUYBE inefficient because we call this function in a loop and call dictFetchValue every time.
-    struct RedisModule *module = dictFetchValue(modules,modulename);
+    if (modulename != cached_modulename) {
```

Comment:
there is a possibility you'll get the same string pointer pointing to a different string (malloc returning the same pointer)
but also, with this solution you don't cache the repeated call to `ACLGetCommandCategoryFlagByName`.
i think you should add some caching capabilities to the commandListFilter struct in some way.